File: app/service/contracts/window.js
windowFactory = function() {
    function create(id, pid, title) {
        if(id === undefined || id === null)
           throw "Parameter id is not set";

        if(pid === undefined || pid === null)
            throw "Parameter pid is not set";

        if(title === undefined || title === null)
            throw "Parameter title is not set";

        return {
            id : id,
            pid : pid,
            icons : {
                image : {},
                imageSelected : {}
            },
            window : {
                title : title
            }
        };
    }

    return {
        create : create
    };
}();

module.exports = windowFactory;

File: src/api/model/window.go
package model

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type WindowResponse struct {
	Windows []Window `json:"windows"`
}

type Window struct {
	Parent     *primitive.ObjectID `bson:"parent" json:"-"`
	Id         int                 `json:"id"`
	Pid        int                 `json:"pid"`
	Window     WindowMetaInfo      `json:"window"`
	Content    *Content            `json:"content,omitempty"`
	ChildIcons *[]Icon             `json:"ChildIcons,omitempty"`
}

type WindowMetaInfo struct {
	Title string `json:"title"`
}

type Icon struct {
	Id            int    `json:"id"`
	Title         string `json:"title"`
	Image         Image  `json:"image"`
	ImageSelected Image  `json:"imageSelected"`
}

type Image struct {
	File   string `json:"file"`
	Width  int    `json:"width"`
	Height int    `json:"height"`
}
